### PR TITLE
[libcoro] Update to 0.10

### PR DESCRIPTION
**Pull request by chausner**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libcoro/vcpkg.json [hunk 1]
```diff
@@ -1,21 +1,19 @@
 {
   "name": "libcoro",
-  "version": "0.9",
+  "version": "0.10",
   "description": "C++20 coroutine library",
   "homepage": "https://github.com/jbaldwin/libcoro",
   "license": "Apache-2.0",
-  "supports": "!windows",
+  "supports": "!(windows & !static)",
```
Comment (dg0yt) [inline ports/libcoro/vcpkg.json:7]: Can't it build windows/dynamic with 
~~~cmake
if(VCPKG_TARGET_IS_WINDOWS)
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
endif()
~~~

Comment (chausner) [reply ports/libcoro/vcpkg.json]: I followed your suggestion in 966b50a4f8f2c6cbeccda1e560cfc99dcbda0dcc. Thanks!

**Comment by Cheney-W** [comment]
```
find_package(PkgConfig REQUIRED)
pkg_check_modules(libcoro REQUIRED IMPORTED_TARGET libcoro)
target_link_libraries(main PkgConfig::libcoro)
```
Above usage failed with below error:
```
Imported target "PkgConfig::libcoro" includes non-existent path
  "include"
in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
* The path was deleted, renamed, or moved to another location.
* An install or uninstall procedure did not complete successfully.
* The installation package was faulty and references files it does not
provide.
```
Also, the contents below in the `libcoro.pc` file will cause the library to be not found on Windows. 
```
Libs: "-L${libdir}" -lcoro
```
This requires modifying the corresponding part in the CMakeLists.txt file.
```
# generate pc file for pkg-config
string(REGEX REPLACE "^lib" "" target1 ${PROJECT_NAME})
configure_file(libcoro.pc.in libcoro.pc @ONLY)
```

**Comment by chausner** [comment]
@Cheney-W Thanks for validating pkgconfig. I had some difficulties testing pkgconfig on Windows but I attempted to fix the issue in b4b75f120252b0d218cbe51ce70f6d4b27c15932. Please give it another try and let me know if it works now.

**Comment by Cheney-W** [comment]
New pc file works fine.

**Review by Cheney-W — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**